lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -919,6 +920,59 @@ impl LocalShard {
     pub async fn shard_recovery_point(&self) -> RecoveryPoint {
         self.clock_map.lock().await.to_recovery_point()
     }
+
+    /// Resolve the WAL delta on the current shard for the given `recovery_point`
+    ///
+    /// The delta can be sent over to the node which the recovery point is from, to restore its
+    /// WAL making it consistent with the current shard.
+    ///
+    /// On success, a WAL record number from which the delta is resolved is returned.
+    /// If a WAL delta could not be resolved, an error is returned describing the failure.
+    pub async fn resolve_wal_delta(
```

Comment:
we might want to decouple this fuction from it's implementation and `self`, cause we might want to test it independently from LocalShard (which requires a lot of hustle to init properly)